### ISSUE TITLE
Risingwave is not able to parse the some INTERVAL query

### ISSUE THREAD

**Issue by sushantgupta** [risingwavelabs/risingwave#7059]
### Describe the bug

In PostgreSQL:
```
todwvwql=> SELECT '1 mons 1 days 00:00:01.000001'::INTERVAL; 
        interval
-------------------------
 +0-1 +1 +0:00:01.000001
(1 row)

todwvwql=> SELECT '1 mons 1 days 00:00:00.000001'::INTERVAL; 
        interval
-------------------------
 +0-1 +1 +0:00:00.000001
(1 row)

todwvwql=>
```

In Rising wave:

```
dev=> SELECT '1 mons 1 days 00:00:01.000001'::INTERVAL; --working fine
       ?column?
----------------------
 1 mon 1 day 00:00:01
(1 row)

dev=> SELECT '1 mons 1 days 00:00:00.000001'::INTERVAL; --here is the issue
ERROR:  QueryError: Expr error: Parse error: risingwave_common::types::interval::IntervalUnit
dev=>
```

**Comment by fuyufjh** [comment]
@ZENOTME Can you please take a look today? We hope to fix this in release-0.1.15

**Comment by BowenXiao1999** [comment]
Draft a pr that should close this for now: https://github.com/risingwavelabs/risingwave/pull/7096

however, The value will lost because we use miliseconds in IntervalUnit, this one is microseconds level. we should fix this after #4514, but now seems it is untracked in our plan.
